File: backend/src/main/scala/microservices/train/objects/BookTrainItemPlannerResponse.scala
package com.typesafe.travel.train.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class BookTrainItemPlannerResponse(orderId: String, orderItemId: String)
object BookTrainItemPlannerResponse:
  given sourceEncoder: Encoder[BookTrainItemPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[BookTrainItemPlannerResponse] = deriveDecoder

File: frontend/src/microservices/train/objects/BookTrainItemPlannerResponse.ts
// 本文件定义 train 模块的 `BookTrainItemPlannerResponse`，用于表示订票结果并提供 JSON 编解码。

export type BookTrainItemPlannerResponse = {
  orderId: string
  orderItemId: string
}
export const bookTrainItemPlannerResponseFromJson = (json: string): BookTrainItemPlannerResponse =>
  JSON.parse(json) as BookTrainItemPlannerResponse

export const bookTrainItemPlannerResponseToJson = (value: BookTrainItemPlannerResponse): string =>
  JSON.stringify(value)
